[CS2113-F11-1] FITNUS

MrOPPA1 opened the pull request.

MrOPPA1 committed:
Merge branch 'master' into master

MrOPPA1 committed:
Merge pull request #14 from MrOPPA1/master

Meal Logging

Remy9926 committed:
Merge branch 'master' into exercise-log-class

Remy9926 committed:
Merge pull request #11 from Remy9926/exercise-log-class

Add ExerciseLog class

MrOPPA1 committed:
fix bugs

Remy9926 committed:
Implement ExerciseLog into Duke

Remy9926 committed:
Resolve Merge Conflict in Parser Class

Remy9926 committed:
Remove dummy assert false statement

Remy9926 committed:
Refactor Parser Class for LogCommand handling

Remy9926 committed:
Remove unused import

Remy9926 committed:
Fix ViewLogCommand not displaying how amount of exercises logged

Remy9926 committed:
Merge pull request #18 from Remy9926/exercise-log-class

Implement ExerciseLog into Duke

Remy9926 committed:
Fix checkstyle errors

Remy9926 committed:
Fix checkstyle errors

Remy9926 committed:
Merge pull request #19 from Remy9926/fix-checkstyle-errors

Fix checkstyle errors

Remy9926 committed:
Enable assertions in build.gradle

Remy9926 committed:
Merge pull request #20 from Remy9926/enable-assertions

Enable assertions in build.gradle

Remy9926 committed:
Add comments to ExerciseLog and associated classes

Remy9926 committed:
Merge pull request #21 from Remy9926/exercise-log-class

Add comments to ExerciseLog and associated classes

Remy9926 committed:
Update DG with Log update feature (#22)

MrOPPA1 committed:
fix meal

MrOPPA1 committed:
Merge pull request #27 from MrOPPA1/master

fix meal

kvn-zhang committed:
Update DeveloperGuide.md

Remy9926 committed:
Fix errors (#29)

Fix errors

Remy9926 committed:
Add update functionality to ExerciseLog class (#30)

Allow users to update exercises within the exercise log

Remy9926 committed:
Add log and update commands to UG

itayrefaely reviewed (commented):
1. First, it feels like there are text blocks in the DG which are maybe too long, not organized in bullet points, and are not very technical. Is it possible that this makes it more difficult to understand?

<img width="1049" alt="Screenshot 2023-11-01 at 12 18 46" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/131960045/a125f158-6136-4ccc-b4fe-d8656b46c045">

2. The DG is missing a few key components, such as target user profile, value proposition, and more.

<img width="496" alt="Screenshot 2023-11-01 at 12 19 58" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/131960045/89dd0fde-4fe7-4b31-b4bd-f5bcc8119a0f">

<img width="843" alt="Screenshot 2023-11-01 at 12 21 15" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/131960045/92d85802-01cb-4116-b3f8-60b11ec1876f">

3. The user stories for versions v1.0 and v2.0 are not related to what appears in the rest of the DG.

4. It seems that the approach you took to make a command superclass is very well explained in the DG, including the logic/reasoning, and also it's implementation structure.

nihalzp commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,37 @@
+
+Below is a class diagram that describes how our various components interact with one another:
+
+![image](https://github.com/AY2324S1-CS2113-F11-1/tp/assets/54857388/21bd2aa9-8e50-49d2-b888-50446115f922)
```
In the class diagram, maybe use same font size for the text of same type?

yicheng-toh replied:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/75836313/7b9ccbc6-36c2-424f-b5b4-9f4bc5599283)
Should it be  in:Scanner instead of Scanner in?

yicheng-toh replied:
Good use of '+' and '-' for the class diagram.

yicheng-toh replied:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/75836313/4f1e510d-cc7f-4592-af3d-a1fea7c6b363)

Should there be a line instead of spaces between the attributes and the methods?

yicheng-toh replied:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/75836313/fc7608c1-c5ff-4094-b221-aca4c8f1c7a5)

Perhaps the arrow should be a closed triangle?
Good use of dotted arrow for the implementation of interface.

nihalzp commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,37 @@
+
+![image](https://github.com/AY2324S1-CS2113-F11-1/tp/assets/54857388/21bd2aa9-8e50-49d2-b888-50446115f922)
+
+The parser class is responsible for reading the user input and creating specific command objects. All of these subcommands inherit from a general Command parent class which has an `execute` method that can be called regardless of child class and will return a CommandResult object. The CommandResult object dictates what info (string or list of items) needs to be returned to the user for viewing. B ...
```
```suggestion
The parser class is responsible for reading the user input and creating specific command objects. All of these subcommands inherit from a general `Command` parent class which has an `execute` method that can be called regardless of child class and will return a `CommandResult` object. The `CommandResult` object dictates what info (string or list of items) needs to be returned to the user for viewing. Both `CommandResult` and `TextUI` (responsible for actually displaying data from `CommandResult` to user) will only print objects from classes that inherit from the `Printable` interface. This ensures that objects in the `relevantItems` list are all serializable as a human-readable string to be displayed to the end user.
```
Maybe enclose the class/interface names in code-blocks?

Hongzhii commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,37 @@
+
+<img width="652" alt="Screenshot 2023-10-22 at 22 25 09" src="https://github.com/Remy9926/tp/assets/95456114/d07d5570-11fc-426c-877d-feb31a338f0a">
 
 ## Product scope
```
unfinished DG

Hongzhii commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,37 @@
+
+Step 3: The user calls the update command with the information of the old exercise as well as the new information that the user wants to update with. With this, the update is done.
+
+<img width="652" alt="Screenshot 2023-10-22 at 22 25 09" src="https://github.com/Remy9926/tp/assets/95456114/d07d5570-11fc-426c-877d-feb31a338f0a">
```
maybe add a class diagram for the log class

Hongzhii replied:
add a sequence diagram for creating a new log

yicheng-toh reviewed (commented):
Generally well done for DG. Take note of the minor issues.

nihalzp commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,37 @@
+<img width="652" alt="Screenshot 2023-10-22 at 22 25 09" src="https://github.com/Remy9926/tp/assets/95456114/d07d5570-11fc-426c-877d-feb31a338f0a">
 
 ## Product scope
 ### Target user profile
```
Target user profile, Value proposition, User Stories, and Non-Functional Requirements is not updated.

nihalzp commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,37 @@
+
+Below is an example of how this feature will work:
+
+Step 1: Upon starting the app, the ExerciseLog class is initialized and a log is created for the user to log their exercises.
+
+<img width="661" alt="Screenshot 2023-10-22 at 22 21 04" src="https://github.com/Remy9926/tp/assets/95456114/933b3636-eba7-442b-bcf2-aec22ef49dba">
+
+Step 2: The user calls the log command with the specified parameters to add a new exercise to the list. However, the user notices that they made a typo and want to change the details of the exercise that they just logged.
```
```suggestion
- **Step 1:** Upon starting the app, the ExerciseLog class is initialized and a log is created for the user to log their exercises.

<img width="661" alt="Screenshot 2023-10-22 at 22 21 04" src="https://github.com/Remy9926/tp/assets/95456114/933b3636-eba7-442b-bcf2-aec22ef49dba">

- **Step 2:** The user calls the log command with the specified parameters to add a new exercise to the list. However, the user notices that they made a typo and want to change the details of the exercise that they just logged.
```
Maybe indent and bold the steps section to make it create a hierarchy of sections.

nihalzp reviewed (commented):
Good job on making progress on the DG with diagrams and explanations!

kvn-zhang committed:
Update UserGuide.md

J-Y-Yan committed:
add goal functuon content

J-Y-Yan committed:
Update DeveloperGuide.md

J-Y-Yan committed:
Update DeveloperGuide.md

J-Y-Yan committed:
Update DeveloperGuide.md

J-Y-Yan committed:
Update DeveloperGuide.md

J-Y-Yan committed:
Update DeveloperGuide.md

J-Y-Yan committed:
Update DeveloperGuide.md

J-Y-Yan committed:
Add documentation

J-Y-Yan committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-F11-1/tp

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

MrOPPA1 committed:
update exception handler

MrOPPA1 committed:
update user guide

MrOPPA1 committed:
Merge pull request #31 from AY2324S1-CS2113-F11-1/meal

Meal

J-Y-Yan committed:
add deleteGoal command

Remy9926 committed:
Fix checkstyle errors (#32)

Remy9926 committed:
Standardize assert method used (#33)

Remy9926 committed:
Update Parser.java

Remy9926 committed:
Throw exceptions in ExerciseLog and associated classes (#35)

Remy9926 committed:
Implement viewing exercises in ExerciseLog (#36)

Remy9926 committed:
Fix PED bugs (#60)

* Throw exceptions when input is not properly formatted

* Implement viewing exercises in ExerciseLog

* Handle exceptions in ExerciseLog with details explaining what went wrong

Remy9926 committed:
Continue fixing PED issues (#72)

* Throw exceptions when input is not properly formatted

* Implement viewing exercises in ExerciseLog

* Handle exceptions in ExerciseLog with details explaining what went wrong

* Update UG to include updated update command

* Add commands to Command Summary

* Add StorageFile handling in Duke

* Add writing to file in LogCommand class

* Add functionality to delete exercises from file

* Implement functionality for updating exercises from text file

Remy9926 committed:
Add Remy9926 PPP (#73)

* Add Remy9926 PPP

* Refine documentation

Remy9926 committed:
Implement additional features + information (#74)

* Throw exceptions when input is not properly formatted

* Implement viewing exercises in ExerciseLog

* Handle exceptions in ExerciseLog with details explaining what went wrong

* Fix checkstyle error

* Fix issue with indices in ExerciseLog Add better exception handling to ViewLogCommand

* Add hasExercise method

* Update UG to include updated update command

* Add commands to Command Summary

* Initialize StorageFile Class

* Add StorageFile handling in Duke

* Add writing to file in LogCommand class

* Add functionality to delete exercises from file

* Implement functionality for updating exercises from text file

* Add Remy9926 PPP

* Refine documentation

* Update Help Command

J-Y-Yan committed:
add delete goal function

J-Y-Yan committed:
remove unused class and correct indentation issues

J-Y-Yan committed:
Correct indentation issues

J-Y-Yan committed:
Correct indentation errors and give more detailed documentation

J-Y-Yan committed:
Correct indentation errors

J-Y-Yan committed:
Merge pull request #75 from AY2324S1-CS2113-F11-1/deleteGoal

Delete goal functionality

J-Y-Yan committed:
Add view Goal list funcationality

J-Y-Yan committed:
Merge pull request #76 from AY2324S1-CS2113-F11-1/viewGoal

Add view Goal list functionality

J-Y-Yan committed:
Update AboutUs.md

J-Y-Yan committed:
Update AboutUs.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Allow more format for user inputs

J-Y-Yan committed:
Add achieve goal functionality

J-Y-Yan committed:
Correct symbol for decode

J-Y-Yan committed:
Merge pull request #81 from AY2324S1-CS2113-F11-1/achieveGoal

Add achieve goal functionality

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update the help message

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Add view achievement functionality

J-Y-Yan committed:
Correct Indentations

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Merge pull request #82 from AY2324S1-CS2113-F11-1/viewAchievement

Add view achievement functionality

kvn-zhang committed:
PPP

kvn-zhang committed:
Merge pull request #83 from kevinz420/master

Add Kevin Zhang's PPP

J-Y-Yan committed:
Bug fix in addGoal command execution

J-Y-Yan committed:
Merge pull request #84 from AY2324S1-CS2113-F11-1/viewAchievement

Bug fix in add goal command

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Correct feedback to user of delete goal method

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Add date parse function for goal object

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Add date format conversion from user input to MMM d, yyyy

J-Y-Yan committed:
Merge pull request #87 from AY2324S1-CS2113-F11-1/timePraserOfGoal

Add date parse function for goal object

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Rename variable storage and file to exerciseLogStorage and exerciseLogFile to prevent confusion

J-Y-Yan committed:
Add FileIO function to restore saved goal record

J-Y-Yan committed:
Bug fix in loading saved data successfully but clear data in save file accidentally once at the start of app.

J-Y-Yan committed:
Add documentation for Date class

J-Y-Yan committed:
Correct all checkStyle errors, including indentations, word limit in line, etc.

J-Y-Yan committed:
Merge pull request #90 from AY2324S1-CS2113-F11-1/goalStorage

Add FileIO function to restore saved goal record

J-Y-Yan committed:
Change all path filed into constant and correct some warnings in using static field as instance field

J-Y-Yan committed:
Add FiloIO for achievement record

J-Y-Yan committed:
Merge pull request #91 from AY2324S1-CS2113-F11-1/heads/FileIO-achievement

Add FiloIO for achievement record

J-Y-Yan committed:
Correct variables name to fullfill check style requirements

MrOPPA1 committed:
rewrite date and dateTime class
implement meal_delete class
implement meal_list class
add private attribute time to meal class

MrOPPA1 committed:
Merge pull request #94 from AY2324S1-CS2113-F11-1/meal

rewrite date and dateTime class

MrOPPA1 committed:
Add category enum into Meal class.
Fix the incorrect index of DeleteCommand.
Show total calories in the ListCommand.
Fix incorrect List type in MealCommand.
Add DataManager to manage the file io of the project data.

MrOPPA1 committed:
Delete the output of saving files.
Creating deliverables V2.0.

MrOPPA1 committed:
Merge pull request #95 from AY2324S1-CS2113-F11-1/meal

Meal

MrOPPA1 committed:
Rewrite user guide.

MrOPPA1 committed:
Merge pull request #96 from AY2324S1-CS2113-F11-1/meal

Rewrite user guide

MrOPPA1 committed:
Rewrite AboutUs.md .

MrOPPA1 committed:
Merge pull request #97 from AY2324S1-CS2113-F11-1/meal

Rewrite AboutUs.md .

J-Y-Yan committed:
Create j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Remove IncorrectDateException class

Remy9926 committed:
Change exerciselog command names (#92)

* Change exerciselog-related command words

* Create Storage as abstract class

* Update documentation with newest exerciselog commands

Remy9926 committed:
Kevinz420 bug fixes (#98)

* Fixed code review suggestions

* Added various bug fixes

* Resolve Parser class conflicts

* Remove StorageFile Class

Remy9926 committed:
Bug fixes (#99)

* Fixed code review suggestions

* Added various bug fixes

Remy9926 committed:
Bug fixes (#100) (#101)

v2.1

J-Y-Yan committed:
Remove redundant content on table

J-Y-Yan committed:
Recover GoalCommand class

J-Y-Yan committed:
Remove redundant comment block on duke class

J-Y-Yan committed:
Recover GoalStorage Class

J-Y-Yan committed:
Recover AchmStorage Class

J-Y-Yan committed:
Recover Duke class

J-Y-Yan committed:
Recover Duke class

J-Y-Yan committed:
Recover Duke Class

J-Y-Yan committed:
Recover Praser

J-Y-Yan committed:
Correct Storage class

J-Y-Yan committed:
Add goal package and goal list class

J-Y-Yan committed:
Recover TextUi class

J-Y-Yan committed:
Recover all goal command executions

J-Y-Yan committed:
Move Goal class into goal package

J-Y-Yan committed:
Remove the abstraction of Storage Class

J-Y-Yan committed:
Correct documentations

J-Y-Yan committed:
Recover the passed time detect funtion of Date Class

J-Y-Yan committed:
Update DeveloperGuide.md

J-Y-Yan committed:
Update DeveloperGuide.md

J-Y-Yan committed:
Change date time class to allow or disallow past detect

J-Y-Yan committed:
Simplify if-else condition

J-Y-Yan committed:
Correct GoalList import issues

MrOPPA1 committed:
Add Project Portfolio Page.
Supplement the DeveloperGuide.md .

J-Y-Yan committed:
Correct all documentation issues, including
- return of various methods
- exception of various methods
- param of various methods

MrOPPA1 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-F11-1/tp

kvn-zhang committed:
Update kevinz420.md

MrOPPA1 committed:
doc

MrOPPA1 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-F11-1/tp

MrOPPA1 committed:
Fix error

Remy9926 committed:
Bug fixes (#103)

* Fixed code review suggestions

* Added various bug fixes

* Add exit command handling

MrOPPA1 committed:
Fix Category enum.
Fix CategoryParser class.

MrOPPA1 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-F11-1/tp

MrOPPA1 committed:
Fix Category enum.
Fix CategoryParser class.

MrOPPA1 committed:
fix bugs

Fix the bugs that unable to see the category enum and category parse class.

MrOPPA1 committed:
Merge pull request #104 from AY2324S1-CS2113-F11-1/meal_

fix bugs

J-Y-Yan committed:
Complete help message for goal related class

J-Y-Yan committed:
Redundant change

J-Y-Yan committed:
Recover storage for goal

MrOPPA1 committed:
fix missing

fix missing

MrOPPA1 committed:
Merge branch 'master' into meal_

MrOPPA1 committed:
Merge pull request #105 from AY2324S1-CS2113-F11-1/meal_

fix missing

J-Y-Yan committed:
Update mroppa1.md

J-Y-Yan committed:
Update j-y-yan.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

MrOPPA1 committed:
PR-V2.1

MrOPPA1 committed:
Merge pull request #107 from AY2324S1-CS2113-F11-1/meal_

PR-V2.1

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md

J-Y-Yan committed:
Update UserGuide.md
